src/DenseACSets.jl

```diff
@@ -149,31 +196,32 @@ macro abstract_acset_type(head)
     _ => (head, GlobalRef(DenseACSets, :StructACSet))
   end
   esc(quote
-    abstract type $type{S,Ts} <: $parent{S,Ts} end
+    abstract type $type{S,Ts,P} <: $parent{S,Ts,P} end
   end)
 end
 
 """ This is a SimpleACSet which has the schema as a field value rather
 than as a type parameter.
 """
-struct DynamicACSet <: SimpleACSet
+struct DynamicACSet{PT} <: SimpleACSet{PT}
```

Comment:
This seems like you are using PT as the part type, rather than as the implementation strategy marker as in `StructACSet`.